src/kmeans.jl

```diff
@@ -230,37 +234,34 @@ function update_assignments!(
     end
 end
 
-#
-#  Update centers based on updated assignments
-#
-#  (specific to the case where samples are not weighted)
-#
-function update_centers!(
-    X::AbstractMatrix{T},            # in: sample matrix (d x n)
-    w::Nothing,                      # in: sample weights
-    assignments::Vector{Int},        # in: assignments (n)
-    to_update::Vector{Bool},         # in: whether a center needs update (k)
-    centers::AbstractMatrix{T},      # out: updated centers (d x k)
-    cweights::Vector{Float64}        # out: updated cluster weights (k)
-    ) where T<:Real
-
+####
+#### update centers (unweighted case: weights=nothing)
+####
+
+function update_centers!(X::AbstractMatrix{TX},     # in: sample matrix (p x n)
+                         weights::Nothing,          # in: sample weights
+                         assignments::Vector{Int},  # in: assignments (n)
+                         to_update::Vector{Bool},   # in: whether a center needs update (k)
+                         centers::Matrix{TC},       # out: updated centers (p x k)
+                         cweights::Vector{Int}      # out: updated cluster weights (k)
+                         ) where {TX<:Real,TC<:AbstractFloat}
     d, n = size(X)
     k = size(centers, 2)
 
     # initialize center weights
-    cweights[to_update] .= 0.0
+    cweights[to_update] .= 0
 
     # accumulate columns
-    @inbounds for j = 1:n
+    @inbounds for j ∈ 1:n
```

Comment:
Is `∈` mentioned somewhere as recommended style? FWIW, I prefer `in` over `=`, but `∈` unnecessarily introduces Unicode requirement.

Reply:
ok!